demo/c/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.4)
+project(porcupine_demo_c)
+
+set(CMAKE_C_STANDARD 99)
+
+list(APPEND COMMON_LIBS dl)
```

Comment:
since we are only assign to these variables once, it's better to use `set` instead of `list` and `append`

```
set(COMMON_LIBS dl)
set(MIC_LIBS pthread m)
```